octopusdeploy/schema_tenant_project_environment.go

```diff
@@ -0,0 +1,27 @@
+package octopusdeploy
+
+import "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+func getTenantProjectEnvironmentSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"tenant_id": {
+			Description: "The tenant ID associated with this tenant.",
+			Required:    true,
+			Type:        schema.TypeString,
+			ForceNew:    true,
+		},
+		"project_id": {
+			Description: "The project ID associated with this tenant.",
+			Required:    true,
+			Type:        schema.TypeString,
+			ForceNew:    true,
+		},
+		"environment_id": {
```

Comment:
You can technically link a tenant with no environments. Arguably we should support this scenario by not making the `environment_id` required.

![image](https://github.com/user-attachments/assets/83e6f63a-dd84-46f9-b88a-0197d4028f3c)

Reply:
I didn't think that was possible, which is surprising as I probably worked on the original API :D . Sounds reasonable though if supported by UI.